job_export.py

```diff
@@ -50,14 +50,25 @@
     filters = json.dumps({"filter": input_data.get("filter", {})})
     case_ids = gql.execute(filters=filters)
 
-    with open("/peregrine-creds.json") as pelican_creds_file:
-        peregrine_creds = json.load(pelican_creds_file)
-
-    DB_URL = "jdbc:postgresql://{}/{}".format(
-        peregrine_creds["db_host"], peregrine_creds["db_database"]
-    )
-    DB_USER = peregrine_creds["db_username"]
-    DB_PASS = peregrine_creds["db_password"]
+    try:
+        with open("/peregrine-creds.json") as pelican_creds_file:
+            peregrine_creds = json.load(pelican_creds_file)
+    except (FileNotFoundError, json.JSONDecodeError) as e:
+        print(f"Failed to load credentials file: {e}")
+        peregrine_creds = {}
+
+    # Set variables, prioritizing environment variables
+    DB_HOST = os.getenv("DB_HOST", peregrine_creds.get("db_host"))
+    DB_DATABASE = os.getenv("DB_DATABASE", peregrine_creds.get("db_database"))
+    DB_USER = os.getenv("DB_USER", peregrine_creds.get("db_username"))
+    DB_PASS = os.getenv("DB_PASS", peregrine_creds.get("db_password"))
+
+    # Construct the database URL if possible
+    if DB_HOST and DB_DATABASE:
+        DB_URL = f"jdbc:postgresql://{DB_HOST}/{DB_DATABASE}"
+    else:
+        DB_URL = None
+        print("DB_HOST or DB_DATABASE is missing. DB_URL cannot be constructed.")
```

Comment:
Should we raise an error in this scenario? 🤔  So the application doesn't continue thinking everything is ok, but it has no database. Not sure what we usually do for that in other services.

Reply:
Yeah if no db exists then the job should throw runtime error